code/controllers/controllers.js

```diff
@@ -59,4 +62,61 @@ exports.pushArticleVotes = (req, res, next) => {
     .catch((err) => {
       next(err);
     });
-};
\ No newline at end of file
+};
+
+exports.postComment = (req, res, next) => {
+  //Firstly, need to check of article exists before trying to insert a comment fo that article
+  selectOneArticle(req)
+    .then((result) => {
+      // Secondly, need to check user exists before trying to insert a comment
+      selectOneUser(req)
+        .then((result) => {
+          // now safe to insert comment because user exists
```

Comment:
SQL will already throw is an error if we try to invoke this with a user that doesn't exist.

Reply:
done - have now stripped out my bespoke error handling and put PSQL error handling in